src/map_ref.rs

```diff
@@ -0,0 +1,214 @@
+use crate::iter::*;
+use crate::HashMap;
+use crossbeam_epoch::{self as epoch, Guard};
+use std::borrow::Borrow;
+use std::collections::hash_map::RandomState;
+use std::fmt::{self, Debug, Formatter};
+use std::hash::{BuildHasher, Hash};
+use std::ops::Index;
+
+/// A reference to a [`HashMap`], constructed with [`HashMap::as_ref`].
+/// The current thread will be pinned for the duration of this reference.
```

Comment:
I think it is worthwhile to document the implications of that here, since they are pretty severe. Specifically, no destructors will be run for as long as this lives!